Assignment4/Assignment4_Guillehf01/Task07.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# **Task 07: Querying RDF(s)**
+
+# In[1]:
+
+
+get_ipython().system('pip install rdflib')
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+
+# First let's read the RDF file
+
+# In[2]:
+
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# In[3]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+contRDFLib = 0
+contSPARQL = 0
+
+#RDFLib
+print("RDFLib Query:")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+    contRDFLib+=1
+    print(f'Subclass {contRDFLib}: {s}')
+
+print("\n")
+print("SPARQL Query:")
+#SPARQL
+q1 = prepareQuery('''
+  SELECT ?Subclass WHERE { 
+    ?Subclass rdfs:subClassOf ns:LivingThing.
+  }
+  ''',
+  initNs = {"ns":ns, "rdfs":RDFS}
+)
+
+for r in g.query(q1):
+    contSPARQL+=1
+    print(f'Subclass {contSPARQL}: {r.Subclass}')
+
```

Comment:
In both exercises you don't retrieve the subclasses of subclasses of living person